modules/general_functions.py

```diff
@@ -656,3 +657,51 @@ def find_next(iterable: Iterable[T], cond: Callable[[T], bool]) -> T:
         return next(i for i in iterable if cond(i))
     except StopIteration:
         raise ValueError("Value not found in iterable.")
+
+# FIXME: Find better placement
+class ReferenceDensity:
+    def __init__(self, layers: List = None):
+        self.layers = layers
+        self.reference_density = 0.0
+        self.layer_reference_density = 0.0
+        self.reference_density_limit = 10.0
+        self.reference_density_meter = 10.0
+        self.masses: List = []
+        self.g_mass = 0.0
+
+    def update_reference_density(self):
+        for layer in self.layers:
+            self.masses = []
+            # If there is only one layer
+            if len(self.layers) == 1:
+                self.calculate_reference_density(layer)
+
+            # If the layer's width is smaller than 10 nm
+            elif self.reference_density_limit - layer.thickness >= 0:
+                self.reference_density_meter = self.reference_density_meter - \
+                                          layer.thickness
```

Comment:
Use augmented assignment:

```suggestion
                self.reference_density_meter += layer.thickness
```

Reply:
OK